File: programmers/lv2/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    days = [math.ceil((100 - a) / b) for a, b in zip(progresses, speeds)]
    
    now = 0
    cnt = 1
    for next in range(1, len(days)):
        if days[now] >= days[next]:
            cnt += 1
        else:
            now = next
            answer.append(cnt)
            cnt = 1
    answer.append(len(days) - now)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> results = new ArrayList<>();
        Queue<Integer> queue = new LinkedList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int day = (int) Math.ceil((100 - progresses[i]) / (double) speeds[i]);
            queue.offer(day);
        }
        
        while(!queue.isEmpty()) {
            int curDay = queue.poll();
            int count = 1;
           while (!queue.isEmpty() && queue.peek() <= curDay) {
                queue.poll();
                count++;
            }

            results.add(count);
        }

        
        return results.stream().mapToInt(i -> i).toArray();
    }
}